modules/core/src/main/java/org/apache/ignite/internal/lock/ReusableLockLockup.java

```diff
@@ -29,15 +29,20 @@
 public class ReusableLockLockup implements AutoLockup {
     private final Lock lock;
 
-    public ReusableLockLockup(Lock lock) {
+    private ReusableLockLockup(Lock lock) {
         this.lock = lock;
     }
 
+    public static ReusableLockLockup forLock(Lock lock) {
```

Comment:
`new ReusableLockLockup(lock)` is confusing because the reader might get an impression that the lock is actually acquired. I switched to a static method to have ability to specify with method name that it's not acquired here. Probably `forLock()` is not sufficient here.

How about `bind()`? This would give a hint that we just wrap the lock, without acquiring it.

Reply:
I think that constructor implies wrapping, not changing of the state. Usually constructors don't do anything other than setting fields, especially in a class as simple as this